frontend/ui/size.lua

```diff
@@ -26,7 +26,14 @@ defaults, please take a second to consider:
 ]]
 
 local dbg = require("dbg")
-local Screen = require("device").screen
+local Device = require("device")
+local Screen = Device.screen
+
+-- This one is required for Size.max_items below,
+-- because generic device does not have Device.display_dpi.
+if Device.display_dpi == nil then
+    Device.display_dpi = 1
+end
```

Comment:
Don't poke at constant fields, and especially don't randomly *write* to them ;).

If you truly really always want the native display dpi, use a public method for it, and just implement a sane fallback in your callers here.

Reply:
Is my comment above a better approach? https://github.com/koreader/koreader/pull/11234#issuecomment-1859366840

Reply:
Nope, just use `Device:getDeviceScreenDPI() or 1`